[Shurvir Arora] iP

shurvirarora opened the pull request:
## Yale
> "Failure is an option here. If things are not failing, you are not innovating enough." – Elon Musk ([source](https://www.inc.com/larry-kim/50-innovation-amp;-success-quotes-from-spacex-founder-elon-musk.html))

Yale frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

You can retrieve its source code from [here.](https://github.com/shurvirarora/ip)

Yale is able to,

1. Add tasks (with _dates_)
2. Delete tasks
3. Mark/Unmark them as done.
4. Find them based on your entered keyword :100:

And it is **FREE**!

Upcoming features for next release:
 - [x] Use Gradle
 - [ ] Use Checkstyle
 - [ ] GUI

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    new Yale().run();
}
```

Commit:
Add Gradle support

shurvirarora committed:
Implement skeletal version of Yale with greet, echo & exit feature

shurvirarora committed:
Create ToDoList class and ToDoItem class
Store command text in ToDoList, display back to user when requested
Change function produceOutput() to performAction()
Remove echo() function

ToDoList : List Structure to contain multiple ToDoItem
ToDoItem : Represents information of a particular task

Reason: Easier to implement more features and behaviours to do with the tasks.

shurvirarora committed:
Add new method getItem() in ToDoList class
Add new boolean attribute "isMarked" to ToDoItem class
Add new methods getStatusIcon(), markItem(), unmarkItem() within ToDoItem class

Reason: To abstract the information required so as to ease method calls in main()
and to mark and unmark the tasks.

shurvirarora committed:
Add chatbot response for unmarkItem()
Clean up long lines for better viewing of code

shurvirarora committed:
Naming convention: Change class name from ToDoItem to Task, change class name from ToDoList to TaskList

shurvirarora committed:
Add ToDo, Deadline, Event classes which are subclasses of Task,
Add class variable of Task called totalTasks to keep track of total number of tasks

shurvirarora committed:
Split classes into their own respective java files

shurvirarora committed:
Create package yalebot, Improve toString() of classes Deadline, Event, Task,
Modify performAction() in Yale to cover for "Deadline", "Event" and "ToDo" cases,
Add some edge cases to account for marking/unmarking number that doesnt exist,
Modify addTo() function in TaskList: adds an item of type Task,
Modify listOut() function in TaskList: formats list response message correctly

shurvirarora committed:
Remove totalTasks variable since list.getSize() is used to retrieve the total number of tasks

shurvirarora committed:
Remove YALE logo due to line break errors in testing, add commands to input.txt, add expected outputs to EXPECTED.txt

shurvirarora committed:
Add IndexOutOfBoundsException to mark/unmark, deadline, event and todo features,
Add NumberFormatException to mark/unmark feature,
Add error message if anything invalid is entered

shurvirarora committed:
Add Delete feature along with NumberFormatException and
IndexOutOfBoundsException in Yale.java

TaskList class: Add deleteItem() method

shurvirarora committed:
Add more input commands in input.txt for comprehensive testing

shurvirarora committed:
Improve deadline and event feature in Yale.java

shurvirarora committed:
Add Javadoc comments to all files

shurvirarora committed:
Modify expected output for UI testing

shurvirarora committed:
Add data/yale.txt as a text file to write data to and retrieve data from

shurvirarora committed:
Change package name from yalebot to yale.task

shurvirarora committed:
Create class FileWrite and implement writeToFile() method

Reason: To write data into a data/yale.txt

shurvirarora committed:
Add a feature that writes the tasks into data/yale.txt

shurvirarora committed:
class Yale: Add writeActionTo() method to save changes in tasks list to data/yale.txt

shurvirarora committed:
Create class FileRead to read data from data/yale.txt

shurvirarora committed:
class Yale: Add readSavedData() method to read saved tasks from data/yale.txt

shurvirarora committed:
Improve chatbot response when there are saved tasks and no saved tasks

shurvirarora committed:
Update JavaDoc for class FileRead, FileWrite and Yale

shurvirarora committed:
Implement export() method to write structured data to file data/yale.txt

shurvirarora committed:
class FileRead: Change printFileContents() to loadFileContents() to structure file data into a String
class TaskList: Implement exportOut() and importIn() to export and import data from specified file respectively
class Yale: Call loadFileContents() and importIn() to load data as soon as Yale starts up

shurvirarora committed:
Handle folder-does-not-exist-yet case

shurvirarora committed:
Store deadline dates as objects instead of Strings
Reformat date output

shurvirarora committed:
Merge branch 'branch-Level-7'

shurvirarora committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Yale.java
#	src/main/java/yale/task/Deadline.java
#	src/main/java/yale/task/Event.java

# Solved merge conflicts

shurvirarora committed:
Ignore files in data

shurvirarora committed:
Update README.md

shurvirarora committed:
Create class: Parser to deal with user command
Implement performAction() in class: Parser

shurvirarora committed:
Remove performAction() method from Yale.java
Implement method calls dealing with list, delete, mark, todo, deadline, event features in class: TaskList

shurvirarora committed:
Merge commit '155bb122ed90fa9996044ef601667f4735afa124'
Update README.md

shurvirarora committed:
Create class: Ui to deal with interactions with the user

shurvirarora committed:
Create Storage class to deal with loading tasks from and saving tasks in specified file

shurvirarora committed:
Remove methods from class: FileWrite and class: FileRead

Reason: These methods have been shift to class: Storage

shurvirarora committed:
Remove class: FileRead and class: FileWrite

Reason: The contents of these classes have been shift to class: Storage, hence there is no need for these classes

shurvirarora committed:
Move checkExit() method from class: Yale to class: Ui

Reason: Ui class will handle interaction with user input

shurvirarora committed:
Improve print response in markTask() and unmarkTask() methods

shurvirarora committed:
Create yale.command package

shurvirarora committed:
Create test directory
Create TaskListTest class

shurvirarora committed:
Add 3 test methods for TaskListTest class

shurvirarora committed:
Add getSize() test method to TaskListTest class

shurvirarora committed:
Create TaskTest class

shurvirarora committed:
Add 5 test methods to TaskTest class

shurvirarora committed:
Modify loadFileContents() in Storage class

Reason: To account for case where data folder or file path do not exist

shurvirarora committed:
Create JAR file Yale.jar

shurvirarora committed:
Implement findFeature() in TaskList class to give users a way to find a task

shurvirarora committed:
TaskList class: Improve conditional statement in findFeature() method

shurvirarora committed:
class TaskList: Add JavaDoc

shurvirarora committed:
class Parser: Add JavaDoc

shurvirarora committed:
Add JavaDoc for Task.java and Storage.java

shurvirarora committed:
Add JavaDoc for Deadline.java, Event.java and ToDo.java

shurvirarora committed:
Add JavaDoc for Yale.java, Parser.java and Ui.java

shurvirarora committed:
Merge branch 'branch-Level-9'

shurvirarora committed:
Tweak code to comply with coding standard

shurvirarora committed:
Merge branch 'branch-A-JavaDoc'

shurvirarora committed:
Merge branch 'branch-A-CodingStandard'

shurvirarora committed:
class Yale: Create Yale() and run() methods in order to organise main() method

shurvirarora committed:
class Yale: Add JavaDoc

raynerljm commented on src/main/java/Yale.java:
```diff
@@ -0,0 +1,55 @@
+
+import java.util.Scanner;
+
+/**
```
Good job with the JavaDocs!

raynerljm commented on src/main/java/Yale.java:
```diff
@@ -0,0 +1,55 @@
+     * @param args
+     */
+
+    private static final String FILE_PATH = "data/yale.txt";
```
Good job with naming constants with screaming snake case

raynerljm commented on src/main/java/yale/command/Parser.java:
```diff
@@ -0,0 +1,44 @@
+     * @param command
+     * @param list
+     */
+    public void performAction(String command, TaskList list) {
```
Good job naming your methods a verb

raynerljm commented on src/main/java/yale/command/Parser.java:
```diff
@@ -0,0 +1,44 @@
+            list.findFeature(command, list);
+        } else {
+            System.out.println("Error: " + command
+                    + " is not a valid command. Please try again.");
```
Good job starting a new line with an operator

raynerljm commented on src/main/java/yale/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+     * Boolean isMarked to determine
+     * if tasked is marked
+     */
+    protected boolean isMarked = false;
```
Good job with the boolean naming convention

raynerljm commented on src/main/java/yale/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    @Override
+    public String toString() {
+        return getStatusIcon() + " " + this.name;
```
Slight note, you don't have to use 'this.name' when there is no ambiguity, you can just use 'name'. It's an optional coding standard

shurvirarora replied:
Noted on this, thanks!

raynerljm commented on src/main/java/yale/task/TaskList.java:
```diff
@@ -0,0 +1,265 @@
+     * ArrayList that can contain
+     * elements of type Task
+     */
+    ArrayList<Task> list;
```
Might be good to name this variable something like tasks - a plural variable name

raynerljm commented on src/main/java/yale/command/Parser.java:
```diff
@@ -0,0 +1,44 @@
+
+import yale.task.TaskList;
+
+/**
```
Note you should add a fullstop after each line in JavaDocs

shurvirarora replied:
Rectified this. Thanks!

raynerljm reviewed (approved):
LGTM! Great job

shurvirarora committed:
Update JavaDocs

shurvirarora committed:
Add Gradle Support
Set up Checkstyle

shurvirarora committed:
Merge branch 'add-gradle-support' of https://github.com/shurvirarora/ip into add-gradle-support

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

shurvirarora committed:
Amend Checkstyle violations

shurvirarora committed:
Merge branch 'add-gradle-support'

shurvirarora committed:
Update JavaDoc

shurvirarora committed:
Update build.gradle to include JavaFX
Create Launcher class
Test Launcher by writing psuedo start() method in Yale.java

shurvirarora committed:
class Yale: Update start() method, add getDialogLabel() to improve GUI

shurvirarora committed:
Create DialogBox class to improve GUI

shurvirarora committed:
Add images User.png and Yale.png to improve GUI

shurvirarora committed:
Update DialogBox class and incorporate it in Yale.java

shurvirarora committed:
Add DialogBox.fxml and MainWindow.fxml

shurvirarora committed:
Add Main.java, MainWindow.java to update GUI

shurvirarora committed:
Improve Ui class, Add close window feature for GUI in MainWindow.java

shurvirarora committed:
class Yale: Update getResponse() to improve GUI
class Parser: Change print statements to return Strings in performAction()

shurvirarora committed:
Create package gui
Import DialogBox, Launcher, Main, MainWindow into package gui

shurvirarora committed:
Improve GUI

shurvirarora committed:
Merge branch 'branch-Level-10'

shurvirarora committed:
Update GUI response message

shurvirarora committed:
Amend Checkstyle violations

shurvirarora committed:
Improve GUI appearance

Shivang211 commented on src/main/java/yale/task/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+     * @param isMarked
+     * @param by
+     */
+    public Deadline(String name, boolean isMarked, String by) {
```
Good job starting a boolean variable with 'is' 👍

Shivang211 commented on test/yale/task/TaskTest.java:
```diff
@@ -0,0 +1,47 @@
+class TaskTest {
+
+    @Test
+    void getStatusIcon_isMarked() {
```
Avoid using underscores for function naming, use camel case convention

Shivang211 commented on src/main/java/Yale.java:
```diff
@@ -0,0 +1,55 @@
+        ui.welcomePrompt();
+        Scanner scanner = new Scanner(System.in);
+        String fileData = storage.loadFileContents();
+        list.importIn(fileData);
```
Can assign this expression to a boolean, instead of writing it indepedantly

shurvirarora committed:
class Task: Create getter method getName() to retrieve Task name

shurvirarora committed:
class Task: Add assertion in getName() method
Add assert statement in build.gradle

shurvirarora committed:
class TaskList: Add assertion in deleteFeature() method

shurvirarora committed:
class Deadline, class Event: Add assertion in export() method

shurvirarora committed:
class TaskList: Improve readability of listOut(), exportOut(), listFeature(), findFeature() methods

There are complicated expression in the conditional statements of these methods that can be written better to improve readability.

Removing these complicated expressions and rephrasing them would make the method easier to read.

As a step towards better readability, let's extract the complicated expressions and store them as variables, which can then be used in the conditional statements. Doing so will improve readability of the methods.

shurvirarora committed:
class Storage: Improve readability of loadFileContents() method

shurvirarora committed:
Merge pull request #3 from shurvirarora/branch-A-Assertions

Branch a assertions

shurvirarora committed:
Merge pull request #4 from shurvirarora/branch-A-CodeQuality

Branch a code quality

shurvirarora committed:
Merge branch 'master' of https://github.com/shurvirarora/ip

shurvirarora committed:
class Storage: Improve loadFileContents() method to load stored data from data/yale.txt

shurvirarora committed:
Create DoWithinPeriod class

In order to add a feature that manages tasks that need to be done within a certain period, a new class DoWithinPeriod needs to be created.

Creating this class as a subclass of Task allows it to have a is-a relationship with Task, thereby inheriting the attributes and methods from the Task class.

As a result, DoWithinPeriod is a type of Task object with its own modification, just like Event, Deadline and Todo.

shurvirarora committed:
Add startPeriod, endPeriod attributes, Modify constructor, export() and toString() methods

For the DoWithinPeriod task, we require a starting date and ending date of the period, hence we created the startPeriod and endPeriod attributes.

In order for a DoWithinPeriod object to behave uniquely, we have to override the export() and toString() method from Task.

Doing so will create a unique DoWithinPeriod object, fulfilling its purpose and still abiding as a Task object.

shurvirarora committed:
Update JavaDocs

shurvirarora committed:
class DoWithinPeriod: Update constructor method, export(), toString() methods

shurvirarora committed:
class TaskList: Update importIn() method, Create doWithinPeriodFeature() method

In order to incorporate the DoWithinPeriod task, we have to create a doWithinPeriodFeature that handles the user input for that case and creates a DoWithinPeriod object and adds it into the list.

The importIn() method needs to be updated to incorporate for the format that the DoWithinPeriod object is written into data/yale.txt file. We can then extract the data from yale.txt containing the DoWithinPeriod task and add it to the list such that its data is saved every time we close the program.

Doing so would allow for the DoWithinPeriod object to be saved and loaded automatically as the program is started.

shurvirarora committed:
class Parser: Add doWithinPeriodFeature() in performAction() method

This is to ensure that every time the program is required to perform an action, it accounts for the doWithinPeriodFeature() as well.

shurvirarora committed:
class TaskList: Improve doWithinPeriodFeature() method

shurvirarora committed:
Merge pull request #5 from shurvirarora/branch-B-DoWithinPeriodTasks

Branch b do within period tasks

shurvirarora committed:
Merge branch 'master' of https://github.com/shurvirarora/ip

shurvirarora committed:
class Yale: Update JavaDocs

shurvirarora committed:
class TaskList: Add clearTaskList() and clearListFeature() methods
class Parser: Add clearListFeature() into performAction() method

This feature allows to clear the list immediately, which is useful as it is faster than deleting each task one by one.

shurvirarora committed:
Improve TextField and Send button of GUI

shurvirarora committed:
Improve GUI font response for marking and unmarking tasks

shurvirarora committed:
Change Yale.png picture for better GUI

shurvirarora committed:
Improve GUI and response format

shurvirarora committed:
class Ui: Include GUI response in welcomePrompt() when there are no saved tasks

shurvirarora committed:
class TaskList: Add exception handling for findFeature()

shurvirarora committed:
class Main: Add Yale icon to GUI

shurvirarora committed:
Add Ui.png, Improve GUI styling

shurvirarora committed:
Improve GUI response

shurvirarora committed:
Update README.md

shurvirarora committed:
Amend checkstyleMain errors

shurvirarora committed:
Set theme jekyll-theme-architect

shurvirarora committed:
Set theme jekyll-theme-cayman

shurvirarora committed:
Set theme jekyll-theme-cayman

shurvirarora committed:
Update README.md

shurvirarora committed:
Update README.md

shurvirarora committed:
Update README.md

shurvirarora committed:
Update README.md

shurvirarora committed:
Update README.md

shurvirarora committed:
Modify build.gradle for building of JAR file
